packages/clerk-js/src/core/clerk.ts

```diff
@@ -907,7 +907,7 @@ export class Clerk implements ClerkInterface {
       beforeUnloadTracker?.stopTracking();
     }
 
-    if (redirectUrl) {
+    if (redirectUrl && !beforeEmit) {
```

Comment:
`Clerk.signOut()` uses both `beforeEmit` and `redirectUrl` which is an incorrect state as the redirection should be trigger by only one of them.

Reply:
Should we fix signOut? Then throw in this if both beforeEmit and redirectUrl come through.